### ISSUE TITLE
variable plot with ellipses grouped using colors

### ISSUE THREAD

**Issue by Olifair** [kassambara/factoextra#139]
Hello, 
I'm using ade4 and dudi.pca to do my analysis. 

When I do a plot of individuals, with ellipses of my four groups, then this works fine : 
fviz_pca_ind(pc2,
             col.ind = datpca$group, # color by groups
             palette = c("blue",  "red","lightblue","darkorange"),
             addEllipses = TRUE, # Concentration ellipses
             ellipse.type = "confidence",
             legend.title = "Groups",
             repel = TRUE
)

where datpca is the dataframe used for the pca analysis (group variable is excluded), and the group is a factor with four levels. 

However when I do : 
fviz_pca_var(pc2,
             col.var = datpca$group, # color by groups
             palette = c("blue",  "red","lightblue","darkorange"),
             addEllipses = TRUE, # Concentration ellipses
             ellipse.type = "confidence",
             legend.title = "Groups",
             repel = TRUE
)

I get the following error : 

Error in fviz(X, element = "var", axes = axes, geom = geom.var, color = col.var,  : 
  The length of color variableshould be the same as the number of rows in the data.

The vector "datpca$group" is the same dataframe, so the length of the data is the same : there is no missing data in the data.frame. the PCA output "pc2" contains in $tab 12 observations, exactly as in the group vector. And anyway, it works for individuals. 
 I have found similar questions on the web in different forums, but no answers.  

I would be very happy if somebody could provide me with a way to work around this problem
Thanks in advance for your time

**Comment by Nelson-Gon** [comment]
I am having a similar issue with `fviz_pca_var`, did you solve it? I think there's some bug in the internal working of length matching for `fviz_pca_var`. The `.add_ind_groups` function which is the source of this error does work fine independently.

**Comment by Olifair** [comment]
@Nelson-Gon  No, sorry, I had to work around this problem. Haven't found a solution since neither. it seems that @kassambara is not active on this package anymore.

**Comment by nweerasu** [comment]
Bump, in case. Still having a similar issue

**Comment by kassambara** [comment]
This isn't a bug — it's a conceptual mismatch, and the error message was just unclear (now improved, see below).

`fviz_pca_var()` colours the **variables** (the arrows), so `col.var` must have one value **per variable** (or be a metric like `"cos2"`/`"contrib"`). You passed `datpca$group`, which has one value **per observation** — that length matches the individuals, not the 12 variables, hence the error. Ellipses around variables grouped by an observation-level grouping also isn't defined.

What you want is exactly your **`fviz_pca_ind()`** call — group the individuals and draw ellipses (that already works). Variables are coloured per-variable, e.g. by a metric:

``` r
suppressWarnings(suppressPackageStartupMessages(library(factoextra)))
res <- prcomp(iris[, -5], scale. = TRUE)

# What you want: colour/group the INDIVIDUALS by your factor, with ellipses.
# (col.ind has one value per observation; pick any ellipse.type.)
fviz_pca_ind(res, col.ind = iris$Species, geom = "point",
             addEllipses = TRUE, ellipse.type = "norm",
             legend.title = "Groups")
#> Ignoring unknown labels:
#> • linetype : "Groups"
```

![](https://i.imgur.com/ICH5R8u.png)

``` r

# Variables are coloured per VARIABLE (one value per column), e.g. by a metric:
fviz_pca_var(res, col.var = "contrib")
```

![](https://i.imgur.com/XOEuv9L.png)

<sup>Created on 2026-06-25 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

(Works the same with your ade4 `dudi.pca` object.) I also clarified the error message in PR #235 so it now explains the length must match the elements being plotted — individuals for `fviz_*_ind()`, variables for `fviz_*_var()` — or be a metric. Closing as resolved (not a bug).
